### PR TITLE
fix: allow assets to open from preview view

### PR DESCRIPTION
**Pull request by nickolay-kondratyev**
# fix: allow assets to open from preview view

### Quality Assurance
- [x] Add a [test](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#writing-tests) for the new feature
- [ ] Make sure all the existing [tests](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#running-all-tests) pass
- [x] Do a spot check by running your feature with our [test workspace](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#test-workspace)
- [x] If your tests involve running [[manual Testing|dendron://dendron.dendron-site/dendron.dev.qa.test#manual-testing]], make sure to update [[Test Workspace|dendron://dendron.dendron-site/dendron.dev.ref.test-workspace]] with any necessary notes/additions to perform manual test ^9jtWc6ov340p
- [ ] After you submit your pull request, check the output of our [integration test](https://github.com/dendronhq/dendron/actions) and make sure all tests pass
  - NOTE: If you running MacOS or Linux, pay special attention to the Windows output and vice versa if you are developing on Windows

### CODE REVIEW

File: packages/plugin-core/src/commands/ShowPreview.ts [hunk 1]
```diff
@@ -94,31 +102,104 @@ export const extractNoteIdFromHref = (data: {
...
   MARKDOWN = "MARKDOWN",
   UNKNOWN = "UNKNOWN",
 }
 
 const classifyLink = ({ href }: NoteViewMessage["data"]): LinkType => {
   if (href && href.startsWith("vscode-webview")) {
-    return LinkType.WIKI;
-  }
-  if (href && (href.startsWith("http://") || href.startsWith("https://"))) {
+    if (href.includes("/assets/")) {
```
Comment (kevinslin) [inline packages/plugin-core/src/commands/ShowPreview.ts:115]: Assets are referenced via `/assets/`,  `assets/` and `./assets/`. The above would miss the second case. 

There's also an edge case where we could have someone with a vault named assets at which point the cross vault link would look like the following: `dendron://assets/{wiki-link}`

I forget if the vault name shows up as part of the `href`. I like to say no but good to double check

File: test-workspace/dendron.2021.11.22.113354516.migrate-configs.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
+version: 3
+useFMTitle: true
```
Comment (kevinslin) [inline test-workspace/dendron.2021.11.22.113354516.migrate-configs.yml:2]: this is an artifact of #1743 . shouldn't be in the commit

File: test-workspace/dendron.2021.11.22.172645373.migrate-configs.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
+version: 3
+useFMTitle: true
```
Comment (kevinslin) [inline test-workspace/dendron.2021.11.22.172645373.migrate-configs.yml:2]: see above comment

File: test-workspace/vault/dendron.ref.assets.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+---
+id: Hf27I1UR3HvKyd6HRh8C0
+title: Assets
+desc: ''
+updated: 1637576810331
+created: 1637568178090
+---
+
+## PDF pasted with `Dendron: Paste File` command
+[dummy-pdf.pdf](assets/dummy-pdf.pdf)
```
Comment (kevinslin) [inline test-workspace/vault/dendron.ref.assets.md:10]: should also add case of `./assets/` and `/assets/`.

also, can we add a vault or rename one of the local vaults to `assets` and create a cross vault link to it?

**Review by kevinslin — CHANGES_REQUESTED** [review]

File: packages/plugin-core/src/commands/ShowPreview.ts (hunk 1 above)
Comment (nickolay-kondratyev) [reply packages/plugin-core/src/commands/ShowPreview.ts]: Good call out didn't know `dendron://assets/{wiki-link}` was a thing. 

On the different asset formatting it appears by the time it gets to `href` they are all formatted the same added manual testing for that as well. Let me know if you meant some different way to reference the assets than the ones that are added in the manual test:

```
* [dummy-pdf.pdf](assets/dummy-pdf.pdf) 
* [dummy-pdf.pdf](./assets/dummy-pdf.pdf) 
* [dummy-pdf.pdf](/assets/dummy-pdf.pdf) 
```

File: test-workspace/vault/dendron.ref.assets.md (hunk 4 above)
Comment (nickolay-kondratyev) [reply test-workspace/vault/dendron.ref.assets.md]: Added 

New vault `assets` and asset link to it from `vault` file 
```
* [test-file-in-another-vault-which-is-named-assets](dendron://assets/assets/assets/test-file-in-asset-vault.md)
```

File: packages/plugin-core/src/commands/ShowPreview.ts (hunk 1 above)
Comment (kevinslin) [reply packages/plugin-core/src/commands/ShowPreview.ts]: miscommunication here - when i say they could be a vault for assets, I meant someone can create a vault and name it `assets`. 

in that case, you would have `assets` appear in the url but actually not be an asset. so `[[ dendron://assets/foo]]` should open the note foo. if the vault name doesn't show up in the `href` then the point should be moot and we don't need to handle it

File: packages/plugin-core/src/commands/ShowPreview.ts [hunk 5]
```diff
@@ -94,31 +102,138 @@ export const extractNoteIdFromHref = (data: {
...
   WEBSITE = "WEBSITE",
   MARKDOWN = "MARKDOWN",
   UNKNOWN = "UNKNOWN",
 }
 
 const classifyLink = ({ href }: NoteViewMessage["data"]): LinkType => {
-  if (href && href.startsWith("vscode-webview")) {
+  if (
+    href &&
+    (href.startsWith("dendron://") ||
```
Comment (kevinslin) [inline packages/plugin-core/src/commands/ShowPreview.ts:116]: we only support this format for cross vault links to dendron notes, not assets. 
its interesting to think how we would link to assets in other vaults but we'd want to think about how we want to support that. 

we're actually moving away from regular markdown links alltogether and handling assets according to this instead -> [[Note References over Note Types|dendron://private/task.markdown.2021.11.18.note-references-over-note-types]]

**Review by kevinslin — CHANGES_REQUESTED** [review]

File: packages/plugin-core/src/commands/ShowPreview.ts (hunk 5 above)
Comment (nickolay-kondratyev) [reply packages/plugin-core/src/commands/ShowPreview.ts:116]: Rolled back the change of 'dendron://'

**Review by kevinslin — APPROVED** [review]

**Pull request merged**